Fix missing React keys in embedding settings page

nemanjaglumac opened the pull request:
[Resolves: #32395]

How to check:
1. there should be no console error about missing keys
2. sections should have unique test ids
    - `-standalone-embeds-setting`
    - `-full-app-embedding-setting`

nemanjaglumac committed:
Fix missing React keys in embedding settings page

[Resolves: ##32395]

nemanjaglumac commented on frontend/src/metabase/admin/settings/selectors.js:
```diff
@@ -471,13 +471,15 @@ const SECTIONS = {
         getHidden: (_, derivedSettings) => !derivedSettings["enable-embedding"],
       },
       {
+        key: "standalone-embeds",
```
Nit: `standalone-embedding` sounds a bit nicer and goes along nicely with `enable-embedding` and `full-app-embedding`, but I guess we have to follow the name of the section 🤷‍♂️

ranquild replied:
I'm not sure if it's to use `key` as it's meant to represent the actual name of the setting on the backend. Perhaps the rendering code should be aware of missing keys and use something else instead

nemanjaglumac replied:
Just to make sure I understand - are you saying that if we use the `key` here in selectors, it could potentially mess up the communication to the backend?

I understood the second part, which is to set the missing keys in the place where we actually render this code.

WiNloSt replied:
> Just to make sure I understand - are you saying that if we use the key here in selectors, it could potentially mess up the communication to the backend?

I think Alex meant, if he's not sure if `key` in the settings are meant to communicate to readers that it represents BE settings key. e.g.
https://github.com/metabase/metabase/blob/6e14e94f138fb8ec07b3f03e2e5be64917b038a1/frontend/src/metabase/admin/settings/selectors.js#L507-L513

This very same setting is used by the BE.

I think what you should do is to use `-` prefix like other settings that doesn't represent the BE settings. e.g.
https://github.com/metabase/metabase/blob/6e14e94f138fb8ec07b3f03e2e5be64917b038a1/frontend/src/metabase/admin/settings/selectors.js#L514-L519

```suggestion
        key: "-standalone-embeds",
```

This applies to `-full-app-embedding` as well

By the way, I can guarantee it doesn't have anything to do with communicating to our BE in this case, because these 2 settings don't have a way to modify settings in anyway.

nemanjaglumac replied:
@WiNloSt @ranquild done.

This feels weird, tbh.
I would've never guessed what the `-` prefix meant. Maybe we should make this more explicit in the future.

WiNloSt replied:
> This feels weird, tbh. I would've never guessed what the `-` prefix meant. Maybe we should make this more explicit in the future.

I agree, this is what we already have and I figure it's our de facto convention, but I don't have any idea how to achieve that now.

ranquild reviewed (commented):
comments inline

nemanjaglumac committed:
Address code review comments

We're adding prefix `-` to the keys that are not meant to represent backend settings.

WiNloSt reviewed (approved):
Awesome, tested and there is no more key errors.
